node/src/actors/chain_manager/mining.rs

```diff
@@ -863,6 +983,36 @@ fn build_block(
     (block_header, txns)
 }
 
+/// Function that return an Option of a SuperBlock or None in case if there are not blocks
+/// to produce a SuperBlock
```

Comment:
This doc comment is telling me no more than the function signature. And in fact, `None` is still an `Option` of a `SuperBlock` :stuck_out_tongue:  

```suggestion
/// Produces a `SuperBlock` that includes the blocks in `block_headers` if there is at least one of them.
```